pkg/kube/resources.go

```diff
@@ -242,12 +255,28 @@ func CreateResourceProviderFromYaml(yamlContent string) *ResourceProvider {
 }
 
 // CreateResourceProviderFromCluster creates a new ResourceProvider using live data from a cluster
-func CreateResourceProviderFromCluster(ctx context.Context, c conf.Configuration) (*ResourceProvider, error) {
+func CreateResourceProviderFromCluster(ctx context.Context, kubeConfigPath string, c conf.Configuration) (*ResourceProvider, error) {
+	var kubeConf *rest.Config
+	var err error
 	kubeConf, configError := config.GetConfigWithContext(c.KubeContext)
 	if configError != nil {
 		logrus.Errorf("Error fetching KubeConfig: %v", configError)
 		return nil, configError
 	}
+
+	if kubeConfigPath != "" {
+		kubeConf, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+		if err != nil {
+			logrus.Errorf("Error fetching KubeConfig: %v", err)
+			return nil, err
+		}
+	} else {
+		kubeConf, err = config.GetConfig()
+		if err != nil {
+			logrus.Errorf("Error fetching KubeConfig: %v", err)
+			return nil, err
+		}
+	}
```

Comment:
Do you think we can extract all this logic out into a helper function? That way it's not duplicated here and `CreateResourceProviderFromResource`. (Also it looks like we missed the `KubeContext` option there!)

Reply:
KubeContext is specified by specifying kubeconfig, without kubeconfig, KubeContext is meaningless

Reply:
Right exactly. But we're not using `KubeContext` in `CreateResourceProviderFromResource`, even though we have a kubeconfig there.